tests/app.ts

```diff
@@ -29,9 +30,5 @@ export const setupApp = async (
     const _server = app.listen(port, () => resolve(_server))
   })
 
-  return simpleRestProvider(`http://localhost:${port}`) as (
-    type: any,
-    resource: any,
-    params: any
-  ) => any
+  return simpleRestProvider(`http://localhost:${port}`) as DataProvider
```

Comment:
unfortunately, if ra-core forces to install too many dependencies, maybe we can copy the type https://github.com/marmelab/react-admin/blob/master/packages/ra-core/src/types.ts#L87 or key anys here.